unix/vncserver/selinux/vncsession.te

```diff
@@ -32,17 +27,20 @@ can_exec(vnc_session_t, vnc_session_exec_t)
 type vnc_session_var_run_t;
 files_pid_file(vnc_session_var_run_t)
 
+type vnc_home_t;
+userdom_user_home_content(vnc_home_t)
+
 allow vnc_session_t self:capability { chown dac_override dac_read_search fowner kill setgid setuid sys_resource };
 allow vnc_session_t self:process { getcap setexec setrlimit setsched };
 allow vnc_session_t self:fifo_file rw_fifo_file_perms;
 
 allow vnc_session_t vnc_session_var_run_t:file manage_file_perms;
 files_pid_filetrans(vnc_session_t, vnc_session_var_run_t, file)
 
-manage_files_pattern(vnc_session_t, xdm_home_t, xdm_home_t)
-manage_fifo_files_pattern(vnc_session_t, xdm_home_t, xdm_home_t)
-manage_sock_files_pattern(vnc_session_t, xdm_home_t, xdm_home_t)
-manage_lnk_files_pattern(vnc_session_t, xdm_home_t, xdm_home_t)
+manage_files_pattern(vnc_session_t, vnc_home_t, vnc_home_t)
+manage_fifo_files_pattern(vnc_session_t, vnc_home_t, vnc_home_t)
+manage_sock_files_pattern(vnc_session_t, vnc_home_t, vnc_home_t)
+manage_lnk_files_pattern(vnc_session_t, vnc_home_t, vnc_home_t)
```

Comment:
Does this need a migration strategy? These files might already have the type `xdg_home_t` (or `user_home_t`).

Reply:
Right, @grulja is aware of it, we spoke about some transition HOWTO document. A command like this should do the task:
```
restorecon -FRvn /home/*/.vnc
```

Reply:
Most users will probably not be able to figure that out unfortunately. Should it be automated in some stage?

Any precedence for how this has been handled in the past?

Reply:
The only general advice we can give is run `fixfiles -F onboot` and reboot the system. We have a precedence in running restorecon from specfile, but only for individual files or 1 directory. Running restorecon for /home can take hours and should not be in specfile. Additionally, in cases home directories are elsewhere than in /home, we can hardly cover that efficiently.

Reply:
I already mention/advice that users should run restorecon on their vnc config folder, but that was for migration from old Tigervnc to a new one with proper systemd support. I'm hesitated to pick this change into RHEL 8 as it might again break users configuration and I'm thinking about picking just the one fixing https://github.com/TigerVNC/tigervnc/issues/1189. However, I think that CentOS Stream/RHEL9 might be a good candidate for this change so we don't break it again for users with future updates.

Reply:
We here upstream don't have the luxury of multiple versions like that, so this would put us in an uncomfortable situation.

Would it be a big no-no to do this from vncsession or vncserver?